superpeer/handler/MenuHandler.py

```diff
@@ -22,11 +30,126 @@ def serve(self, choice: str) -> None:
 		elif choice == "QUER":
 			pass
 
+		elif choice == "LISTSUPERPEERS":
+
+			shell.print_green('\nList of known peers:')
+
+			if not LocalData.get_super_friends():
+				shell.print_red('You do not know any superpeers.')
+
+			for count, friend in enumerate(LocalData.get_super_friends(), start=1):
+				friend_ip4 = LocalData.get_super_friend_ip4(friend)
+				friend_ip6 = LocalData.get_super_friend_ip6(friend)
+				friend_port = LocalData.get_super_friend_port(friend)
+				shell.print_blue(
+					f'{count}] {friend_ip4} {friend_ip6} {str(friend_port)}')
+
 		elif choice == "LISTPEERS":
-			pass
+
+			try:
+				conn = database.get_connection(db_file)
+				conn.row_factory = database.sqlite3.Row
+
+			except database.Error as e:
+				print(f'Error: {e}')
+				print('The server has encountered an error while trying to serve the request.')
+				return
+
+			try:
+				peer_list = peer_repository.find_all(conn)
+
+				if not peer_list:
+					shell.print_red('You do not know any peers.')
+					conn.close()
+
+				else:
+					for count, peer_row in enumerate(peer_list, start=1):
+						shell.print_blue(f'{count}]' + peer_row['ip'] + peer_row['port'] + '\n')
+
+			except database.Error as e:
+				conn.rollback()
+				conn.close()
+				print(f'Error: {e}')
+				print('The server has encountered an error while trying to serve the request.')
+				return
 
 		elif choice == "LISTFILES":
-			pass
+			try:
+				conn = database.get_connection(db_file)
+				conn.row_factory = database.sqlite3.Row
+
+			except database.Error as e:
+				print(f'Error: {e}')
+				print('The server has encountered an error while trying to serve the request.')
+				return
+
+			try:
+				files = file_repository.find_all(conn)
+
```

Comment:
Ok, poi integra con la sorgente file interna del superpeer (```shared_files``` in LocalData) che dovrebbe avere già implementato @GiovanniFiorini

Reply:
integrato!